From c8600cd3cb28c92e5f3898c3a8a314454be61a4d Mon Sep 17 00:00:00 2001
From: David Sniatkiwicz <davidsn@marvell.com>
Date: Mon, 16 Jan 2017 16:49:09 +0200
Subject: [PATCH 0746/1345] dts: a80x0: describe SPI partition in shared
 A8040-DB device tree

commit  610502f01ac562c105870909ed7948bbee43630c from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

since SPI partition are duplicated among all A8040 setups,
there is no need to describe them for each board setup,
so this patch relocate SPI partition to be described in
the shared board device tree: armada-8040-db.dtsi.

align with nand partitions, and move the declaration of
spi partitions to armada-8040-db.dtsi.

Change-Id: Ib07b45d6790cdff00d31ae6b2e2f13ce81b644bf
Signed-off-by: David Sniatkiwicz <davidsn@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35655
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35737
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-db-A.dts  |   19 ---------
 arch/arm64/boot/dts/marvell/armada-8040-db-A0.dts |   19 ---------
 arch/arm64/boot/dts/marvell/armada-8040-db-B.dts  |   19 ---------
 arch/arm64/boot/dts/marvell/armada-8040-db-C.dts  |   19 ---------
 arch/arm64/boot/dts/marvell/armada-8040-db-E.dts  |   38 ------------------
 arch/arm64/boot/dts/marvell/armada-8040-db.dtsi   |   44 +++++++++++++++++++++
 6 files changed, 44 insertions(+), 114 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
index 017e97b..a888b54 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
@@ -221,25 +221,6 @@
 				status = "okay";
 				pinctrl-names = "default";
 				pinctrl-0 = <&cps_spi1_pins>;
-				spi-flash@0 {
-					#address-cells = <0x1>;
-					#size-cells = <0x1>;
-					compatible = "jedec,spi-nor";
-					reg = <0x0>;
-					spi-max-frequency = <20000000>;
-					partition@0 {
-						label = "boot";
-						reg = <0x0 0x200000>;
-					};
-					partition@200000 {
-						label = "Filesystem";
-						reg = <0x200000 0xd00000>;
-					};
-					partition@d00000 {
-						label = "boot_2nd";
-						reg = <0xf00000 0x100000>;
-					};
-				};
 			};
 		};
 		pcie@0x600000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A0.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A0.dts
index 4bb934e..22b192a 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-A0.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-A0.dts
@@ -82,25 +82,6 @@
 			};
 			spi0: spi@510600 {
 				status = "okay";
-				spi-flash@0 {
-					#address-cells = <0x1>;
-					#size-cells = <0x1>;
-					compatible = "jedec,spi-nor";
-					reg = <0x0>;
-					spi-max-frequency = <20000000>;
-					partition@0 {
-						label = "boot";
-						reg = <0x0 0x200000>;
-					};
-					partition@200000 {
-						label = "Filesystem";
-						reg = <0x200000 0xd00000>;
-					};
-					partition@d00000 {
-						label = "boot_2nd";
-						reg = <0xf00000 0x100000>;
-					};
-				};
 			};
 		};
 	};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
index 7541b6d..acdf5f2 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
@@ -199,25 +199,6 @@
 				status = "okay";
 				pinctrl-names = "default";
 				pinctrl-0 = <&cps_spi1_pins>;
-				spi-flash@0 {
-					#address-cells = <0x1>;
-					#size-cells = <0x1>;
-					compatible = "jedec,spi-nor";
-					reg = <0x0>;
-					spi-max-frequency = <20000000>;
-					partition@0 {
-						label = "boot";
-						reg = <0x0 0x200000>;
-					};
-					partition@200000 {
-						label = "Filesystem";
-						reg = <0x200000 0xd00000>;
-					};
-					partition@d00000 {
-						label = "boot_2nd";
-						reg = <0xf00000 0x100000>;
-					};
-				};
 			};
 		};
 		pcie@0x600000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts
index ce138f5..0cc83d1 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts
@@ -207,25 +207,6 @@
 				status = "okay";
 				pinctrl-names = "default";
 				pinctrl-0 = <&cps_spi1_pins>;
-				spi-flash@0 {
-					#address-cells = <0x1>;
-					#size-cells = <0x1>;
-					compatible = "jedec,spi-nor";
-					reg = <0x0>;
-					spi-max-frequency = <20000000>;
-					partition@0 {
-						label = "boot";
-						reg = <0x0 0x200000>;
-					};
-					partition@200000 {
-						label = "Filesystem";
-						reg = <0x200000 0xd00000>;
-					};
-					partition@d00000 {
-						label = "boot_2nd";
-						reg = <0xf00000 0x100000>;
-					};
-				};
 			};
 			spi@700600 {
 				pinctrl-0 = <&cps_spi0_pins>;
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
index 4edb1e4..a5babc2 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
@@ -70,25 +70,6 @@
 			};
 			spi0: spi@510600 {
 				status = "okay";
-				spi-flash@0 {
-					#address-cells = <0x1>;
-					#size-cells = <0x1>;
-					compatible = "jedec,spi-nor";
-					reg = <0x0>;
-					spi-max-frequency = <20000000>;
-					partition@0 {
-						label = "boot";
-						reg = <0x0 0x200000>;
-					};
-					partition@200000 {
-						label = "Filesystem";
-						reg = <0x200000 0xd00000>;
-					};
-					partition@d00000 {
-						label = "boot_2nd";
-						reg = <0xf00000 0x100000>;
-					};
-				};
 			};
 		};
 	};
@@ -238,25 +219,6 @@
 				status = "okay";
 				pinctrl-names = "default";
 				pinctrl-0 = <&cps_spi1_pins>;
-				spi-flash@0 {
-					#address-cells = <0x1>;
-					#size-cells = <0x1>;
-					compatible = "jedec,spi-nor";
-					reg = <0x0>;
-					spi-max-frequency = <20000000>;
-					partition@0 {
-						label = "boot";
-						reg = <0x0 0x200000>;
-					};
-					partition@200000 {
-						label = "Filesystem";
-						reg = <0x200000 0xd00000>;
-					};
-					partition@d00000 {
-						label = "boot_2nd";
-						reg = <0xf00000 0x100000>;
-					};
-				};
 			};
 		};
 		pcie@0x600000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index 55ab0f6..0c7b81c 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -63,6 +63,28 @@
 				bus-width = <8>;
 				non-removable;
 			};
+			spi0: spi@510600 {
+				status = "disabled";
+				spi-flash@0 {
+					#address-cells = <0x1>;
+					#size-cells = <0x1>;
+					compatible = "jedec,spi-nor";
+					reg = <0x0>;
+					spi-max-frequency = <20000000>;
+					partition@0 {
+						label = "boot";
+						reg = <0x0 0x200000>;
+					};
+					partition@200000 {
+						label = "Filesystem";
+						reg = <0x200000 0xd00000>;
+					};
+					partition@d00000 {
+						label = "boot_2nd";
+						reg = <0xf00000 0x100000>;
+					};
+				};
+			};
 		};
 	};
 
@@ -186,6 +208,28 @@
 					reg = <0x1000000 0x3f000000>;
 				};
 			};
+			spi1: spi@700680 {
+				status = "disabled";
+				spi-flash@0 {
+					#address-cells = <0x1>;
+					#size-cells = <0x1>;
+					compatible = "jedec,spi-nor";
+					reg = <0x0>;
+					spi-max-frequency = <20000000>;
+					partition@0 {
+						label = "boot";
+						reg = <0x0 0x200000>;
+					};
+					partition@200000 {
+						label = "Filesystem";
+						reg = <0x200000 0xd00000>;
+					};
+					partition@d00000 {
+						label = "boot_2nd";
+						reg = <0xf00000 0x100000>;
+					};
+				};
+			};
 		};
 
 		cps_usb3h0_phy: cps_usb3_phy0 {
-- 
1.7.9.5

